src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -37,6 +39,35 @@ impl ScalarFunc {
         let i = try_opt!(self.children[0].eval_int(ctx, row));
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
+
+    #[inline]
+    pub fn ltrim<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let val = try_opt!(self.children[0].eval_string(ctx, row));
+        let pos = val.iter().position(|&x| x != SPACE);
+        if let Some(i) = pos {
+            Ok(Some(Cow::Owned(val[i..].to_vec())))
```

Comment:
Here you can write
```rust
            match val {
                Cow::Borrowed(val) => Ok(Some(Cow::Borrowed(&val[i..]))),
                Cow::Owned(val) => Ok(Some(Cow::Owned(val[i..].to_owned()))),
            }
```
to reduce some mem alloc.

Reply:
OK, thx, I'll refine here